### PR TITLE
Improve bash vs zx vs Execa documentation

### LINKED ISSUE #1099: Document CLI utilities
zx includes many CLI-related utilities: 
  - HTTP download (forwards to `node-fetch-native`)
  - CLI prompts (forwards to `node:readline`)
  - Retrieve `stdin` (built-in logic)
  - Retry failed command (built-in logic)
  - CLI spinner  (built-in logic)
  - Globbing (forwards to `globby`)
  - Finding executable file path (forwards to `which`)
  - List processes (forwards to `@webpod/ps`)
  - Temporary file/directory creation (built-in logic)
  - CLI flags parsing (forwards to `minimist`)
  - ANSI colors (forwards to `chalk`)
  - Filesystem (forwards to `fs-extra`)
  - YAML (forwards to `js-yaml`)

Execa chooses a modular approach instead. Users can pick any CLI-related Node module they want.

We should improve our "zx vs bash vs Bash" documentation to show alternatives to the utilities above.

### PR DESCRIPTION
**Pull request by ehmicky**
Fixes #1099

### CODE REVIEW

File: docs/bash.md [hunk 1]
```diff
@@ -887,6 +1079,160 @@ const {pid} = $`npm run build`;
...
+### CLI arguments
+
+```js
+// zx
+const {myCliFlag} = argv;
+```
+
+```js
+// Execa
+import minimist from 'minimist';
```
Comment (sindresorhus) [inline docs/bash.md:1091]: I would go with https://nodejs.org/api/util.html#utilparseargsconfig

File: docs/bash.md [hunk 2]
```diff
@@ -887,6 +1079,160 @@ const {pid} = $`npm run build`;
...
+```
+
+```js
+// zx
+const files = await glob(['packages/*']);
+```
+
+```js
+// Execa
+import {globby} from 'globby';
```
Comment (sindresorhus) [inline docs/bash.md:1174]: https://nodejs.org/api/fs.html#fspromisesglobpattern-options

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: docs/bash.md (hunk 1 above)
Comment (ehmicky) [reply docs/bash.md]: Fixed in aa3fca9c

File: docs/bash.md (hunk 2 above)
Comment (ehmicky) [reply docs/bash.md]: Fixed in 0cf80f0a

**Pull request merged**